### ISSUE TITLE
Move item within list results in removed item

### ISSUE THREAD

**Issue by Fossil01** [marceljuenemann/angular-drag-and-drop-lists#310]
Anyone care to look why when a item is moved within the USERS list itself the item is removed (should not happen) ? But when items in the ACCESS list get moved within the same list they are not.

https://jsfiddle.net/xrhx3vs3/

**Comment by Fossil01** [comment]
Anyone? @marceljuenemann ?

**Comment by Toufux** [comment]
Did you manage to make it work? I got the same issue

**Comment by Fossil01** [comment]
Nope. No one ever replied here. 

Looking at other libraries next week.

**Comment by Fossil01** [comment]
@DimitriMathot Got this working eventually?

**Comment by DimitriMathot** [comment]
Yes I got this working but I don't have any idea how... I am not even sure that it is me that manage to make it work...
Put your code I could maybe compare with mine

**Comment by Fossil01** [comment]
It's in the first post, the jsfiddle.

**Comment by DimitriMathot** [comment]
I changed  dnd-draggable="user" and I add 
$scope.onMoved = function(list,index) {
    $scope[list].users.splice(index,1); <-- this to splice your array
      $scope[list].users = $scope[list].users.filter(function(item) { return !item.selected; });
    };

And I think it worked

**Comment by Fossil01** [comment]
Cheers, will try that after lunch!

**Comment by Fossil01** [comment]
@DimitriMathot When I use that I cannot move multiple items. Just one will end up on the other side and the rest is removed completely :-(

**Comment by DimitriMathot** [comment]
I did not see that you had 2 dndlist try put the user in both the dnd-draggable and remove the $scope[list].users = $scope[list].users.filter(function(item) { return !item.selected; });

**Comment by Fossil01** [comment]
Still no go :( Hopefully @marceljuenemann will take a look.

**Comment by Fossil01** [comment]
@marceljuenemann ?

**Comment by AlexTimofeyev** [comment]
Hello, remove this lines: 
if (list == "json") {
items.selected = false; 
}
And all will great :)

**Comment by Fossil01** [comment]
No, the items will still disappear when dropping it on the same list as it's dragged from.

**Comment by AlexTimofeyev** [comment]
Hm, strangely, I change method "onDrop"
And, items not removed.

$scope.onDrop = function(list, items, index) {
// Deselect all previously selected items after dropping them in a list
angular.forEach(items, function(item) {
item.selected = false;
});
$scope[list].users = $scope[list].users.slice(0, index)
.concat(items)
.concat($scope[list].users.slice(index));
return true;
}

**Comment by Fossil01** [comment]
Got it now, I removed the whole if else statement.
